### PR TITLE
feat (ui): add vulnerabilities list

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: ui/src/assets/i18n/fr.json [hunk 1]
```diff
@@ -250,6 +251,7 @@
   "common_no_favorites": "Vous n'avez pas encore ajouté de projets ou de workflows dans vos favoris.",
   "common_favorites_all": "Tous les favoris",
   "common_file_too_large": "Votre fichier est trop volumineux (max 100Ko)",
+  "common_vulnerabilities": "Vulnerabilitées",
```
Comment (fsamin) [inline ui/src/assets/i18n/fr.json]: Vulnérabilités

File: ui/src/app/shared/vulnerability/list/vulnerabilities.list.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        </div>
+    </div>
+    <ul>
+        <li *ngFor="let v of filteredVulnerabilities">
+            <div class="header">
+                <div class="dot {{v.severity}}"></div>
+                <div class="text">
+                    <span class="component">{{ ' ' + v.component}}</span>
+                    <span>{{ ' ' + v.version}}</span>
+                    <span class="title">{{ v.title + ' '}}<a [href]="v.link" target="_blank">{{v.cve}}</a></span>
```
Comment (bnjjj) [inline ui/src/app/shared/vulnerability/list/vulnerabilities.list.html]: missing icon that tell us it's an external icon

File: ui/src/app/shared/vulnerability/list/vulnerability.list.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            this.allVulnerabilities = data;
+            this.updateVulns();
+        }
+    };
+    allVulnerabilities: Array<Vulnerability>;
+    filteredVulnerabilities: Array<Vulnerability>;
+
+    updateVulns(): void {
+        if (this.allVulnerabilities) {
+            if (!this.filter || this.filter === '') {
```
Comment (bnjjj) [inline ui/src/app/shared/vulnerability/list/vulnerability.list.component.ts]: only `!this.filter` satisfied also `this.filter === ''`

File: ui/src/app/shared/vulnerability/vulnerabilities.component.ts [hunk 4]
```diff
@@ -70,4 +70,11 @@ export class VulnerabilitiesComponent {
     get vulnerabilities() {
         return this._vulnerabilities;
     }
+
+    filter = '';
+
+    updateFilter(v: Vulnerability): void {
+        this.filter = v.component + ' ' + v.version
+        console.log(this.filter);
```
Comment (bnjjj) [inline ui/src/app/shared/vulnerability/vulnerabilities.component.ts]: to delete

File: ui/src/app/shared/vulnerability/vulnerabilities.html [hunk 5]
```diff
@@ -42,8 +41,16 @@
...
+                 (click)="updateFilter(v)"></div>
         </ng-container>
     </div>
 
+    <div class="list">
+        <app-vulnerabilities-list [filter]="filter" [vulnerabilities]="vulnerabilities"></app-vulnerabilities-list>
+    </div>
+
+</div>
+<div class="ui info message" *ngIf="!vulnerabilities || vulnerabilities === 0">
```
Comment (bnjjj) [inline ui/src/app/shared/vulnerability/vulnerabilities.html]: just `!vulnerabilities` is the same

**Comment by ovh-cds** [comment]
CDS Report build-ui#5282.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -  
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✔ 
  * TU views ✔

**Pull request merged**